### PR TITLE
docs(security) api server authentication rework

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
Rework of API server authentication.

* Removed list of API server functionality, this doc is about auth so we should state reasons to authenticate (admin operations)
* Moved signing token information down below, that should be one of the last things that users learn about
* Split usage to access admin token and generate a new token. Especially when you start with Kuma, you only care about accessing the admin token to move you forward.
* Move exploring token above revocation and signing key to help a user understand the content of the token first.
* Remove comments from commands so they can be copy-pasted
* Applied a bunch of vale suggestions
* Better explanation how admin user is authorized to do everything (say what is the name and group and link to access control)

### CODE REVIEW

**Comment by slonka** [comment]
> Split usage to access admin token and generate a new token. Especially when you start with Kuma, you only care about accessing the admin token to move you forward.

👍

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 1]
```diff
@@ -1,156 +1,167 @@
...
+* The list of groups that a user belongs to
 * Expiration date of the token
 
-The User Token is signed by a signing key that is autogenerated on the control plane. The signing key is SHA256 encrypted.
+### Groups
 
-You can check for the signing key:
+A user can be a part of many groups. Kuma adds two groups to a user automatically:
+* Every authenticated user is a part of `mesh-system:authenticated`.
+* Every user that doesn't provide authentication data is a part of `mesh-system:unauthenticated`.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:20]: Why do we need to explain what unauthenticated (user that doesn't provide authentication) means? Is it ok to just make it simpler by doing something like this:

```suggestion
* authenticated users are a part of `mesh-system:authenticated`.
* unauthenticated users are a part of `mesh-system:unauthenticated`.
```

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 2]
```diff
@@ -1,156 +1,167 @@
...
-
-## Implicit groups
+### Admin user token
 
-A user can be a part of many groups. On top of that, Kuma adds two groups automatically.
-* Every authenticated user is a part of `mesh-system:authenticated`
-* Every user that do not provide authentication data is a part of `mesh-system:unauthenticated`.
-
-### Usage
+Kuma creates an admin user token when the control plane starts.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:24]: Is it different with each start? Is it when it starts or when it's installed?

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 3]
```diff
@@ -1,156 +1,167 @@
...
+The admin user token is a user token issued for user `mesh-system:admin` that belongs to `mesh-system:admin` group.
+This group is [by default authorized](./api-access-control.md) to execute all administrative operations.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:26]:
```suggestion
This group is [authorized by default](./api-access-control.md) to execute all administrative operations.
```

It sounds a bit more natural this way (at least to me) - maybe some native speakers can chip in (@michaelbeaumont @parkanzky).

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 4]
```diff
@@ -228,8 +243,9 @@ If the signing key is compromised, we must rotate it and all the tokens that was
    ::::
 
 2. Regenerate user tokens
-   Create new user tokens. These tokens are automatically created with the signing key that’s assigned the highest serial number, so they’re created with the new signing key.
-   At this point, tokens signed by either new or old signing key are valid.
+
+   Create new user tokens. Tokens are always signed by the signing key with the highest serial number.
+   Now tokens signed by either new or old signing key are valid.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:248]: Using "starting from now" to me more clearly indicates that the action just taken resulted in the change.

```suggestion
   Starting from now tokens signed by either new or old signing key are valid.
```

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 5]
```diff
@@ -346,7 +340,7 @@ All users that provides client certificate are authenticated as user with name `
    mkdir /opt/client-certs
    cp /tmp/tls.crt /opt/client-certs/client1.pem 
    ```
-   All client certificates **MUST** end with `.pem` extension. Remember to only put provide certificates without keys.
+   All client certificates must end with `.pem` extension. Remember to only put provide certificates without keys.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:343]:
```suggestion
   All client certificates must end with `.pem` extension. Remember to only provide certificates without keys.
```

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 6]
```diff
@@ -356,19 +350,21 @@ All users that provides client certificate are authenticated as user with name `
...
 ## Multizone
 
-In multizone setup, the majority of actions are executed on the global control plane.
-However, some actions like generating dataplane tokens are available on zone control planes.
-Authentication credentials are not propagated from global control plane to zone control planes.
-Consistent user tokens across the whole setup can be achieved by manually synchronizing signing key from global to zone control planes. 
+In a multizone setup, users execute a majority of actions on the global control plane.
+However, some actions like generating dataplane tokens are available on the zone control plane.
+The global control plane doesn't propagate authentication credentials to the zone control plane.
+You can set up consistent user tokens across the whole setup can by manually synchronizing signing key from global to zone control planes.
```
Comment (slonka) [inline docs/docs/1.7.x/security/api-server-auth.md:370]: Not a scope of this PR but how do I manually sync that up?

**Review by slonka — CHANGES_REQUESTED** [review]
Overall way better than before, some nitpicks, some things that are clearly wrong. Great job BTW! ❤️

File: docs/docs/1.7.x/security/api-server-auth.md (hunk 6 above)
Comment (lahabana) [inline docs/docs/1.7.x/security/api-server-auth.md:370]:
```suggestion
You can set up consistent user tokens across the whole setup by manually synchronizing signing key from global to zone control planes. 
```

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 7]
```diff
@@ -1,156 +1,167 @@
...
+   ```
 
-In order to generate other user tokens, we need to authenticate as admin. When Kuma starts, it generates admin user token and stores it as a [Global Secret](secrets.md).
+2. Expose Kuma CP to be accessible from your machine
 
-Use `kubectl` to extract the admin token
-```sh
-kubectl get secret admin-user-token -n kuma-system --template={{.data.value}} | base64 -d
-```
+   To access Kuma CP via kumactl, you need to expose Kuma CP outside of a cluster. You can do this in several ways:
```
Comment (lahabana) [inline docs/docs/1.7.x/security/api-server-auth.md:39]:
```suggestion
   To access Kuma CP via kumactl, you need to expose Kuma CP outside of a cluster in one of the following ways:
```

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 8]
```diff
@@ -1,156 +1,167 @@
...
 
-Specify list of revoked IDs separated by `,` and store it as `GlobalSecret` named `user-token-revocations`
+### Token revocation
+
+Kuma doesn't keep the list of issued tokens. To invalidate the token, you can add it to a revocation list.
+Every user token has its own ID. As you saw in the previous section, it's available in payload under `jti` key.
+To revoke tokens, specify list of revoked IDs separated by `,` and store it as `GlobalSecret` named `user-token-revocations`
 
 :::: tabs :options="{ useUrlFragment: false }"
 ::: tab "Kubernetes"
```
Comment (lahabana) [inline docs/docs/1.7.x/security/api-server-auth.md:167]: Maybe in these examples you could use `jwt` .e.g: `pbpaste | jwt decode - --json | jq -r '.payload.jti'`

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 9]
```diff
@@ -244,66 +260,44 @@ If the signing key is compromised, we must rotate it and all the tokens that was
...
+   [Configure a control plane](../documentation/configuration.md) with `KUMA_API_SERVER_AUTHN_TOKENS_BOOTSTRAP_ADMIN_TOKEN` set to `false`.
+   :::
+   ::::
 
-This section describes alternative way of authenticating to API Server.
-Admin client certificates are deprecated. If you are using it, please migrate to user token described above.
+## Admin client certificates
+
+This section describes the alternative way of authenticating to API Server.
+Admin client certificates are deprecated. If you are using it, please migrate to the user token in preceding section.
```
Comment (lahabana) [inline docs/docs/1.7.x/security/api-server-auth.md:296]: Probably should be put in a `::: warning` section

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 10]
```diff
@@ -1,156 +1,167 @@
-# API Server Authentication
+# API server authentication
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply docs/docs/1.7.x/security/api-server-auth.md:1]: It was a `vale` suggestion. I think we should follow it.

File: docs/docs/1.7.x/security/api-server-auth.md [hunk 11]
```diff
@@ -1,156 +1,167 @@
...
-* Accessing GUI
+Kuma exposes API server on [ports](../networking/networking.md) `5681` and `5682` (protected by TLS).
 
-Authenticated user can be authorized to execute administrative actions such as
-* Managing Kuma Secrets (on Universal)
-* Generating Dataplane Tokens
-* Generating User Tokens
+An authenticated user can be authorized to execute administrative actions such as
+* Managing administrative resources like Kuma Secrets on Universal
+* Generating user token, data plane proxy token, zone ingress token, zone token
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply docs/docs/1.7.x/security/api-server-auth.md:7]: tried `Generating different token types .e.g: dpp, zone...` but honestly, it looks weird. I'd rather leave it as is

File: docs/docs/1.7.x/security/api-server-auth.md (hunk 2 above)
Comment (jakubdyszkiewicz) [reply docs/docs/1.7.x/security/api-server-auth.md]: Technically on the first start or the CP, but if you remove it, it's recreated, but I guess recreation part is explained in `Disabling bootstrap of the admin user token`

File: docs/docs/1.7.x/security/api-server-auth.md (hunk 6 above)
Comment (jakubdyszkiewicz) [reply docs/docs/1.7.x/security/api-server-auth.md]: changed synchronizing to copying. Just use kumactl/kubectl to copy global secret from Global CP and put it in Zone CP

**Review by slonka — APPROVED** [review]

**Comment by lahabana** [comment]
@jakubdyszkiewicz should this be merged?

**Pull request merged**